apps/api/src/app/events/services/mail-service/handlers/mailgun.handler.ts

```diff
@@ -0,0 +1,21 @@
+import { MailgunEmailProvider } from '@notifire/mailgun';
+import { ChannelTypeEnum } from '@notifire/shared';
+import { ICredentials } from '@notifire/dal';
+import { BaseHandler } from './base.handler';
+
+export class MailgunHandler extends BaseHandler {
+  constructor() {
+    super('mailgun', ChannelTypeEnum.EMAIL);
+  }
+
+  buildProvider(credentials: ICredentials, from: string) {
+    const config: {
+      apiKey: string;
+      username: string;
+      domain: string;
+      from: string;
+    } = { apiKey: credentials.apiKey, username: null, domain: credentials.domain, from };
```

Comment:
Also here, the username should come from the config
```suggestion
    } = { apiKey: credentials.apiKey, username: credentials.username, domain: credentials.domain, from };
```

Reply:
Yea my bad missed that one as well after updating the ICredentialsDto model.

Reply:
Thanks for noticing!